### PR TITLE
Feature/claudia 5613 script to recover nid

### PR DESCRIPTION
**Pull request by flopezag**
#### Reviewers

@chemaper  @jframos 
#### Description
- Create a class and CLI to obtain the NID of the different Generic Enablers based in the information offered by the FIWARE Catalog.
- Generate the appropriate unit tests to this functionality
- Generate the acceptance tests.
- Include the corresponding files to execute the unit and acceptance tests.
- Correct reference to the wiki page where users can obtain information about the NID of the different Generic Enablers.
- Correct some pep8 errors.
#### Testing

Locally, Unit Tests and Acceptance Tests.

### CODE REVIEW

**Comment by jframos** [comment]
Add _docstrings_ in each implemented class/function in `glancesync/getnid.py` with a brief description of the function and parameters. Update README.rst if it is needed.

**Comment by chemaper** [comment]
What about documentation of the util? It will be in the wiki only or it is a good idea to include something in the README.rst? Probably a brief description of what the command do and a copy&paste of the help of the command is enough.

**Comment by jframos** [comment]
I don't know if `glancesync/getnid.py` is a good place to locate the script because I think it is not part of the core of GlanceSync and it is not used by `sync` command. What do you think of putting the script in `scripts/nid`?

File: glancesync/getnid.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import re
+
+__version__ = '1.0.0'
+
+
+class NID(object):
+    """Class to obtain the different nids values registered in the FIWARE Catalogue.
+    """
+    def __init__(self):
+        self.BASEURL = 'http://catalogue.fiware.org/chapter/'
```
Comment (jframos) [inline glancesync/getnid.py]: What do you think of having a `getnid.cfg` with these configurations (BASEURL, CHAPTER, TYPE)?

File: tests/acceptance/features/nid.feature [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+             | cloud         | cloud-hosting                                          | nids-cloud         |
+             | userinterface | advanced-web-based-user-interface                      | nids-userinterface |
+             | sec           | security                                               | nids-sec           |
+             | iot           | internet-things-services-enablement                    | nids-iot           |
+             | data          | datacontext-management                                 | nids-data          |
+             | apps          | applicationsservices-and-data-delivery                 | nids-apps          |
+
+  @happy_path
+    Scenario Outline: 02: Get the version of the application by invoking the appropriate command line instruction.
+     Given that I have the gitnid application installed
```
Comment (jframos) [inline tests/acceptance/features/nid.feature]: I would rather this step: `Given the getnid application installed`

File: tests/acceptance/features/nid.feature [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+                Examples:
+             | option_value | result_value        |
+             | --version    | Get NID code vx.y.z |
+             | -v           | Get NID code vx.y.z |
+
+  @happy_path
+    Scenario Outline: 03: Get the help information of the application by invoking the appropriate command line instruction.
+     Given that I have the gitnid application installed
+     When  I execute the gitnid application with the option
```
Comment (jframos) [inline tests/acceptance/features/nid.feature]: getnid

File: tests/acceptance/features/steps/nid.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+@when(u'I execute the getnid with the following values')
+def step_obtain_list_ge_nids(context):
+    context.value = dict()
+
+    if context.table is not None:
+        for row in __dataset_utils__.prepare_data(context.table):
+            if 'type_value' in row.headings and row['type_value'] != 'False':
+                type = row['type_value']
+
+                command = '../../glancesync/getnid.py --type {}'.format(type)
```
Comment (jframos) [inline tests/acceptance/features/steps/nid.py]: I think the path to _getnid.py_ script could be in config file too.

File: tests/acceptance/features/steps/nid.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    if context.table is not None:
+        for row in __dataset_utils__.prepare_data(context.table):
+            if 'option_value' in row.headings:
+                option = row['option_value']
+
+                command = '../../glancesync/getnid.py {}'.format(option)
+
+                context.result = cmd(command)
+
+                assert_that(command, is_not(''))
```
Comment (jframos) [inline tests/acceptance/features/steps/nid.py]: I think you mean **context.result**, but the assertion should be executed by   _@ then_   steps.

File: tests/acceptance/features/steps/nid.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+@then(u'the program return the corresponding version of this implementation')
+def step_impl_return_version(context):
+    if context.table is not None:
+        for row in __dataset_utils__.prepare_data(context.table):
+            if 'result_value' in row.headings:
+                expected_result = row['result_value']
+
+                result = __check_version__(context.result, expected_result)
+
+                assert_that(result, is_(True))
```
Comment (jframos) [inline tests/acceptance/features/steps/nid.py]: Custom failure message.

File: tests/acceptance/features/steps/nid.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+
+@then(u'the program return the corresponding help information')
+def step_impl(context):
```
Comment (jframos) [inline tests/acceptance/features/steps/nid.py]: Function name more declarative.

File: tests/acceptance/features/steps/nid.py (hunk 5 above)
Comment (flopezag) [inline tests/acceptance/features/steps/nid.py]: You are right, in this method is better to assert the context.result.

Comment (jframos) [inline tests/acceptance/features/steps/nid.py]: _'@ when'_   steps shouldn't validate results. It should be done by _'@ then'_ steps, which ones are the responsible of the validations.

File: tests/acceptance/features/nid.feature (hunk 3 above)
Comment (flopezag) [inline tests/acceptance/features/nid.feature]: done

File: tests/acceptance/features/nid.feature (hunk 2 above)
Comment (flopezag) [inline tests/acceptance/features/nid.feature]: done

File: glancesync/getnid.py (hunk 1 above)
Comment (flopezag) [inline glancesync/getnid.py]: The question is that there are configuration parameters for the class NID not for the getnid, this is the reason why I include inside the class and not like configuration parameter of the getnid.py

**Comment by flopezag** [comment]
The reason why I put this component inside the glancesync and not inside script is due to I want to add unit tests to the development and I was not sure if I put inside script I can do it.

File: tests/acceptance/features/steps/nid.py (hunk 6 above)
Comment (flopezag) [inline tests/acceptance/features/steps/nid.py]: Done

File: tests/acceptance/features/steps/nid.py (hunk 7 above)
Comment (flopezag) [inline tests/acceptance/features/steps/nid.py]: Done

**Comment by jframos** [comment]
I do not know if `scripts/support` is the suitable location for this script, because all the content of this subdir  is regarding to the new base images and the _Support_ account management.
Perhaps `scripts/getnids/getnid.py` or something like this is a better location.

**Comment by chemaper** [comment]
Only a suggestion about the documentation in the README. When you speak about the purpose of the program in the overview, I think that filling the values of the  wiki page https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/GE-identification should be mentioned.

Anyway, it LGTM

**Comment by chemaper** [comment]
LGTM

**Comment by jframos** [comment]
LGTM

**Pull request merged**